### PR TITLE
API 6.1 for V13

### PR DESCRIPTION
**Pull request by Poolitzer**
Same PR as #3112, just with V13

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -30,6 +30,7 @@ filterwarnings =
 ;    Unfortunately due to https://github.com/pytest-dev/pytest/issues/8343 we can't have this here
 ;    and instead do a trick directly in tests/conftest.py
 ;    ignore::telegram.utils.deprecate.TelegramDeprecationWarning
+markers = dev: If you want to test a specific test, use this
```
Comment (Bibo-Joshi) [inline setup.cfg:33]: please remove

File: telegram/bot.py [hunk 2]
```diff
@@ -3157,6 +3164,8 @@ def set_webhook(
             data['ip_address'] = ip_address
         if drop_pending_updates:
             data['drop_pending_updates'] = drop_pending_updates
+        if secret_token is not None:
+            data["secret_token"] = secret_token
```
Comment (Bibo-Joshi) [inline telegram/bot.py:3168]: codecov says this was not covered

File: setup.cfg (hunk 1 above)
Comment (Poolitzer) [reply setup.cfg:33]: But I like it for development.

Comment (Bibo-Joshi) [reply setup.cfg:33]: we've added it to v20 for that reason. v13 is supposed to get api updates exclusively, so I'd like to stick to that as close as reasonbly possible :)

Comment (Poolitzer) [reply setup.cfg:33]: It's such a small non breaking change though. And it just really eases testing specific tests.

Comment (Bibo-Joshi) [reply setup.cfg:33]: fine

Comment (Poolitzer) [reply setup.cfg:33]: Yes! xD

File: telegram/bot.py [hunk 3]
```diff
@@ -3080,16 +3080,17 @@ def set_webhook(
...
         Use this method to specify a url and receive incoming updates via an outgoing webhook.
         Whenever there is an update for the bot, Telegram will send an HTTPS POST request to the
         specified url, containing a JSON-serialized Update. In case of an unsuccessful request,
         Telegram will give up after a reasonable amount of attempts.
 
-        If you'd like to make sure that the Webhook request comes from Telegram, Telegram
-        recommends using a secret path in the URL, e.g. https://www.example.com/<token>. Since
-        nobody else knows your bot's token, you can be pretty sure it's us.
+        If you'd like to make sure that the Webhook was set by you, you can specify secret data in
+        the parameter :paramref:`secret_token`. If specified, the request will contain a header
```
Comment (harshil21) [inline telegram/bot.py:3092]:
```suggestion
        the parameter ``secret_token``. If specified, the request will contain a header
```
no parameter referencing available in v13, same everywhere below

File: telegram/bot.py [hunk 4]
```diff
@@ -5914,6 +5923,140 @@ def get_chat_menu_button(
...
+        is_flexible: bool = None,
+        timeout: ODVInput[float] = DEFAULT_NONE,
+        api_kwargs: JSONDict = None,
+    ) -> str:
+        """Use this method to create a link for an invoice.
+
+        .. versionadded:: 13.13
+
+        Args:
+            title (:obj:`str`): Product name. :tg-const:`telegram.Invoice.MIN_TITLE_LENGTH`-
```
Comment (harshil21) [inline telegram/bot.py:5957]:
```suggestion
            title (:obj:`str`): Product name. :attr:`telegram.Invoice.MIN_TITLE_LENGTH`-
```
tg-const is not available here, so maybe either use `attr`, or just substitute the number in (hardcode - or should I say _harddoc_)

File: telegram/constants.py [hunk 5]
```diff
@@ -247,7 +247,7 @@
 """
 from typing import List
 
-BOT_API_VERSION: str = '6.0'
+BOT_API_VERSION: str = '6.1'
```
Comment (harshil21) [inline telegram/constants.py:250]: the same change should be done in the docstring above

File: telegram/ext/filters.py [hunk 6]
```diff
@@ -1004,6 +1004,36 @@ def filter(self, message: Message) -> bool:
...
+    """
+
+    class _UserPremium(UpdateFilter):
+        __slots__ = ()
+
+        def filter(self, update: Update) -> bool:
+            return bool(update.effective_user) and bool(update.effective_user.is_premium)
+
+    premium_user = _UserPremium(name="filters.PREMIUM_USER")
+    """This filter filters *any* message that has a
```
Comment (harshil21) [inline telegram/ext/filters.py:1030]:
```suggestion
    """This filter filters *any* message from a
```

File: telegram/ext/filters.py [hunk 7]
```diff
@@ -1004,6 +1004,36 @@ def filter(self, message: Message) -> bool:
     location = _Location()
     """Messages that contain :class:`telegram.Location`."""
 
+    class _UserAttachment(UpdateFilter):
+        __slots__ = ()
+
```
Comment (harshil21) [inline telegram/ext/filters.py:1009]:
```suggestion
        __slots__ = ()
        name = "Filters.user_attachment"

```

File: telegram/ext/filters.py [hunk 8]
```diff
@@ -1004,6 +1004,36 @@ def filter(self, message: Message) -> bool:
...
+        def filter(self, update: Update) -> bool:
+            return bool(update.effective_user) and bool(
+                update.effective_user.added_to_attachment_menu
+            )
+
+    user_attachment = _UserAttachment(name="filters.USER_ATTACHMENT")
```
Comment (harshil21) [inline telegram/ext/filters.py:1015]:
```suggestion
    user_attachment = _UserAttachment()
```

File: telegram/ext/filters.py [hunk 9]
```diff
@@ -1004,6 +1004,36 @@ def filter(self, message: Message) -> bool:
...
+    """This filter filters *any* message that have a user who added the bot to their
+    :attr:`attachment menu <telegram.User.added_to_attachment_menu>` as
+    :attr:`telegram.Update.effective_user`.
+
+    .. versionadded:: 13.13
+    """
+
+    class _UserPremium(UpdateFilter):
+        __slots__ = ()
+
```
Comment (harshil21) [inline telegram/ext/filters.py:1025]:
```suggestion
        __slots__ = ()
        name = "Filters.premium_user"

```

File: telegram/ext/filters.py [hunk 10]
```diff
@@ -1004,6 +1004,36 @@ def filter(self, message: Message) -> bool:
...
+        def filter(self, update: Update) -> bool:
+            return bool(update.effective_user) and bool(update.effective_user.is_premium)
+
+    premium_user = _UserPremium(name="filters.PREMIUM_USER")
```
Comment (harshil21) [inline telegram/ext/filters.py:1029]:
```suggestion
    premium_user = _UserPremium()
```

File: tests/test_invoice.py [hunk 11]
```diff
@@ -195,6 +206,57 @@ def make_assertion(*args, **_):
             protect_content=True,
         )
 
+    @pytest.mark.dev
```
Comment (harshil21) [inline tests/test_invoice.py:209]:
```suggestion
```

Comment (Poolitzer) [reply tests/test_invoice.py:209]: it really keeps coming back

Comment (Bibo-Joshi) [reply tests/test_invoice.py:209]: Sometimes harshil makes me feel like I don't even read while reviewing 🙈 😄

Comment (Poolitzer) [reply tests/test_invoice.py:209]: And me like I can't code :(

Comment (harshil21) [reply tests/test_invoice.py:209]: > it really keeps coming back

we can like, have a git hook which removes that, but its a bit too overkill haha

Comment (Poolitzer) [reply tests/test_invoice.py:209]: true, a bit overkill

File: telegram/constants.py [hunk 12]
```diff
@@ -247,7 +247,7 @@
 """
 from typing import List
 
-BOT_API_VERSION: str = '6.0'
+BOT_API_VERSION: str = '6.2'
```
Comment (harshil21) [inline telegram/constants.py:250]:
```suggestion
BOT_API_VERSION: str = '6.1'
```
the future isn't here.. yet.. 😃

**Review by harshil21 — APPROVED** [review]

**Comment by Bibo-Joshi** [comment]
Why is pre-commit failing? I thought we had managed that in #3027?

**Comment by Poolitzer** [comment]
Is it the same error or was it a different one?

**Comment by Bibo-Joshi** [comment]
On this PR it's

```python
telegram/__main__.py:44: error: unused 'type: ignore' comment
Found 1 error in 1 file (checked 155 source files)
```

on #3027 it was 

```python
telegram/__main__.py:44: error: Module has no attribute "__version__"  [attr-defined]
Found 9 errors in 2 files (checked 155 source files)
```

**Pull request merged**